python/lsst/ts/externalscripts/maintel/random_walk.py

```diff
@@ -0,0 +1,342 @@
+                                 of the rotator in terms of the physical rotator angle (in the
+                                 reference frame of the telescope). Note that the telescope
+                                 will resume tracking the sky rotation.
+
+                    Physical: Select a fixed position for the rotator in the reference frame of
+                              the telescope. Rotator will not track in this mode.
+                type: string
+                enum: ["Sky", "SkyAuto", "Parallactic", "PhysicalSky", "Physical"]
+                default: Physical
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        self.log.debug(
+            f"Setting up configuration: \n"
+            f"  total_time: {config.total_time}\n"
+            f"  radius: {config.radius}\n"
+            f"  min_az: {config.min_az}\n"
+            f"  max_az: {config.max_az}\n"
+            f"  min_el: {config.min_el}\n"
+            f"  max_el: {config.max_el}\n"
+            f"  big_offset_prob: {config.big_offset_prob}\n"
+            f"  big_offset_radius: {config.big_offset_radius}\n"
+            f"  track_for: {config.track_for}\n"
+            f"  rot_value: {config.rot_value}\n"
+            f"  rot_type: {config.rot_type}\n"
+        )
+
+        self.config = config
+
+        self.config.rot_type = getattr(RotType, self.config.rot_type)
+
+        if hasattr(self.config, "ignore"):
+            for comp in self.config.ignore:
+                if comp not in self.tcs.components_attr:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Group. "
+                        f"Must be one of {self.tcs.components_attr}. Ignoring."
+                    )
+                else:
+                    self.log.debug(f"Ignoring component {comp}.")
+                    setattr(self.tcs.check, comp, False)
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+        metadata.duration = self.config.total_time
+
+    async def run(self):
+
+        azel_generator = self.random_walk_azel_by_time(
+            total_time=self.config.total_time,
+            radius=self.config.radius,
+            min_az=self.config.min_az,
+            max_az=self.config.max_az,
+            min_el=self.config.min_el,
+            max_el=self.config.max_el,
+            big_offset_prob=self.config.big_offset_prob,
+            big_offset_radius=self.config.big_offset_radius,
+        )
+
+        for i, az, el in azel_generator:
+            await self.slew_and_track(az, el, target_name=f"random_walk_{i}")
+
+    def random_walk_azel_by_time(
```

Comment:
I really liked how you used the generator here! Great idea!

A couple suggestions, though.

First, consider making this an async generator.

Basically all you need to do is:

```
async def random_walk_azel_by_time(...):
    ...
```

And no other changes are needed in the method itself! Pretty neat!  

But then, you will need to change how your ``run`` method calls it, which brings me to the second suggestion. 

I noticed that you are basically passing ``self.config.something`` as arguments to the method. But all this data is already part of the class, so passing them to the method is kind of an overhead. Instead, consider making ``random_wak_azel_by_time`` have no argument at all. Then, internally, you call directly ``self.config.<parameter>``.

Then, your ``run`` method becomes:

```
         async for i, az, el in self.random_walk_azel_by_time():
             await self.slew_and_track(az, el, target_name=f"random_walk_{i}")
```

Nice and tidy!